python/prog_edu_assistant_tools/prog_edu_assistant_tools/magics.py

```diff
@@ -0,0 +1,186 @@
+import io
+import re
+import sys
+import unittest
+
+from IPython.core.display import HTML
+from IPython.core.magic import (Magics, magics_class, line_magic, cell_magic,
+                                line_cell_magic)
+from jinja2 import Template
+from prog_edu_assistant_tools.summary_test_result import SummaryTestResult
+from pygments.formatters import HtmlFormatter
+from pygments import highlight
+from pygments.lexers import PythonLexer
+from types import SimpleNamespace
+
+
+def autotest(testClass):
+    """Non-magic version of the autotest.
+
+    This function can be used as
+
+        result, log = autotest(MyTestCase)
+
+    """
+    suite = unittest.TestLoader().loadTestsFromTestCase(testClass)
+    errors = io.StringIO()
+    result = unittest.TextTestRunner(verbosity=4,
```

Comment:
Just out of curiosity, what does this level of verbosity give us?

Reply:
Okay, maybe level 4 is not necessary, but without explicitly verbosity of at least 2 the snippets like these are not shown for passing tests:
test_has_hello (HelloOutputTest.HelloOutputTest) ... ok
test_not_empty (HelloOutputTest.HelloOutputTest) ... ok
test_output (HelloOutputTest.HelloOutputTest) ... ok
Compare (verbosity 1):
![image](https://user-images.githubusercontent.com/498073/60097652-bf2c1c00-978e-11e9-92de-5fb58041f3cf.png)
to (verbosity 2):
![image](https://user-images.githubusercontent.com/498073/60097664-c81ced80-978e-11e9-89a0-d5748ed217f0.png)